people/will/homework/hw_1.html

```diff
@@ -7,30 +7,32 @@
 
 <!-- Testing page to try out if and else statements-->
 
+
+<h1 style="text-align: center"> The Ultimate Source for Finding <span style = "color: #f00"> Power Level</span></h1>
+
+
+<form style = "text-align: center; margin-top: 50px;" name = "power1" action = "#">What is your power level? 
+	<input type="text" id = "powerInput" name="text1"><input type="button" name="button1" value="Check power" onclick="powerCheck()">
+</form>
+
+<p style = "margin-top: 200px; text-align: center;">This took <em>far</em> longer than I wish it had.</p>
+
 <script type="text/javascript">
 	
+	var powerLevel = document.getElementById('powerInput');
```

Comment:
@wcholland  `powerLevel` here is defined out of the scope of your function: `powerCheck()`.
This means that you are using `powerLevel` inside of `powerCheck()` but it is not defined there.

This technically will still work, but it is bad practice in general unless you know exactly why this works and choose to "break the rules" so to speak. 

Here's why it will still work:
JavaScript functions are defined with what is called "Lexical Scoping". 

This means that they will run the scope in which they are defined rather than the scope of where they are called... **but** if there is a reference to a variable that's _not_ in their scope, they will look one level up until they find the variable or they reach global scope (in this case `window`).

This can be really tricky when you get into bigger applications, so the moral of the story is: keep your variables in the scope of where they are used :)

I found this article if you are interested in learning more:
http://docstore.mik.ua/orelly/webprog/jscript/ch11_04.htm